cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp

```diff
@@ -21,72 +21,49 @@
-                    <a id="alternative-osf" href="${osfLoginUrl}service=${osfLoginContext.getServiceUrl()}">Sign In</a>&nbsp;&nbsp;&nbsp;&nbsp;
-                </c:when>
-                <c:otherwise>
-                    <c:choose>
-                        <c:when test="${osfLoginContext.isInstitutionLogin()}">
-                            <spring:eval var="osfLoginUrl" expression="@casProperties.getProperty('cas.osf.login.url')" />
-                            <c:choose>
-                                <c:when test="${osfLoginContext.isServiceUrl()}">
-                                    <a id="alternative-osf" href="${osfLoginUrl}service=${osfLoginContext.getServiceUrl()}">Non-institution Login</a>&nbsp;&nbsp;&nbsp;&nbsp;
-                                </c:when>
-                                <c:otherwise>
-                                    <a id="alternative-osf" href="${osfLoginUrl}">Non-institution Login</a>&nbsp;&nbsp;&nbsp;&nbsp;
-                                </c:otherwise>
-                            </c:choose>
-                        </c:when>
-                        <c:otherwise>
-                            <spring:eval var="institutionLoginUrl" expression="@casProperties.getProperty('cas.institution.login.url')" />
-                            <c:choose>
-                                <c:when test="${osfLoginContext.isServiceUrl()}">
-                                    <a id="alternative-institution" href="${institutionLoginUrl}&service=${osfLoginContext.getServiceUrl()}">Login through Your Institution</a>&nbsp;&nbsp;&nbsp;&nbsp;
-                                </c:when>
-                                <c:otherwise>
-                                    <a id="alternative-institution" href="${institutionLoginUrl}">Login through Your Institution</a>&nbsp;&nbsp;&nbsp;&nbsp;
-                                </c:otherwise>
-                            </c:choose>
-                        </c:otherwise>
-                    </c:choose>
-                </c:otherwise>
-            </c:choose>
-            <spring:eval var="osfUrl" expression="@casProperties.getProperty('osf.url')" />
-            <a id="back-to-osf" href="${osfUrl}">Back&nbsp;to&nbsp;OSF</a><br>
-        </div>
-    </c:if>
-
 </div> <!-- END #content -->
 
-<c:if test= "${empty alternativeBottomNone}">
-    <c:if test="${empty alternativeBottomLogout}">
-        <div class="row" style="text-align: center;">
-            <br>
-            <spring:eval var="createAccountUrl" expression="@casProperties.getProperty('osf.createAccount.url')" />
-            <a id="create-account" href="${createAccountUrl}${registeredService.properties.registerUrl.getValue()}">Create Account</a>
-        </div>
-    </c:if>
+<c:if test="${empty serviceParam}">
+    <c:set var="serviceParam" value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
 </c:if>
 
+<div class="bottom-link">
```

Comment:
`bottom.js` now only takes care of the links under the grey login box: Sign In, Sign Out, Create Account, Back to OSF. Each page sets up the flags for what they want. No more complicated logic.